crypto/keys/keys_test.go

```diff
@@ -220,3 +221,68 @@ func TestSignatureAlgorithm(t *testing.T) {
 		}
 	}
 }
+
+func TestGenerateKey(t *testing.T) {
+	for _, test := range []struct {
+		name    string
+		keygen  *keyspb.Generator
+		wantErr bool
+	}{
+		{
+			name: "ECDSA with default params",
+			keygen: &keyspb.Generator{
+				Params: &keyspb.Generator_EcdsaParams{},
+			},
+		},
+		{
+			name: "ECDSA with explicit params",
+			keygen: &keyspb.Generator{
+				Params: &keyspb.Generator_EcdsaParams{
+					EcdsaParams: &keyspb.Generator_ECDSA{
+						Curve: keyspb.Generator_ECDSA_P521,
+					},
+				},
+			},
+		},
+		{
+			name: "RSA with default params",
+			keygen: &keyspb.Generator{
+				Params: &keyspb.Generator_RsaParams{},
+			},
+		},
+		{
+			name: "RSA with explicit params",
+			keygen: &keyspb.Generator{
+				Params: &keyspb.Generator_RsaParams{
+					RsaParams: &keyspb.Generator_RSA{
+						Bits: 4096,
+					},
+				},
+			},
+		},
+		{
+			name:    "No params",
+			keygen:  &keyspb.Generator{},
+			wantErr: true,
+		},
+	} {
+		key, err := NewFromGenerator(test.keygen)
+		if gotErr := err != nil; gotErr != test.wantErr {
+			t.Errorf("%s: NewFromGenerator(_) = (_, %v), want err? %t", test.name, err, test.wantErr)
```

Comment:
nit: Use `%v` instead of `%s` / `%t`. It makes life a lot easier when refactoring code.

nit2: If you don't care about the params to NewFromGenerator, consider omitting them in the message (`"NewFromGenerator() = ..."`).

Ditto for others.